firmware/library/L1_Drivers/onboardled.hpp

```diff
@@ -0,0 +1,31 @@
+// This provides the functions to manipulate the numbered LEDs that are soldered
+// onto the SJTwo Board.
+//
+// The initializaiton function takes care of setting up the pins to function as
+// GPIO outputs. When an output is set to LOW, the LED lights up, when the
+// output is set to HIGH the LED will be off.
+//
+//   For the SJTwo Board:
+//
+//              LED#              PORT#             PHYSICAL PIN#
+//
+//              LED 0      |      P1_01      |      PIN# 75
+//              LED 1      |      P1_08      |      PIN# 73
+//              LED 2      |      P1_10      |      PIN# 71
+//              LED 3      |      P1_15      |      PIN# 69
+
+#pragma once
+
+#include <cstdint>
+
+class OnBoardLedInterface {
+ public:
+           virtual void LEDInit()                                    = 0;
+           virtual void LEDOn(uint8_t led_number)                    = 0;
+           virtual void LEDOff(uint8_t led_number)                   = 0;
+           virtual void LEDSet(uint8_t led_number, bool on)          = 0;
+           virtual void LEDToggle(uint8_t led_number)                = 0;
+           virtual void LEDSetAll(uint8_t value)                     = 0;
+           virtual uint8_t LEDGetValues(void)                        = 0;
+           uint8_t led_states;
```

Comment:
This variable shouldn't be apart of the Interface, but apart of the specific implementation. Also, you put this in the public section, when it should have been in the protected section.